common/src/main/java/io/homeassistant/companion/android/common/data/integration/Entity.kt

```diff
@@ -80,7 +81,25 @@ fun <T> Entity<T>.getFanSpeed(): EntityPosition? {
             max = maxValue
         )
     } catch (e: Exception) {
-        Log.e(EntityExt.TAG, "Unable to get getLightBrightness", e)
+        Log.e(EntityExt.TAG, "Unable to get getFanSpeed", e)
+        null
+    }
+}
+
+fun <T> Entity<T>.getFanSteps(): Int? {
+    return try {
+        if (!supportsFanSetSpeed()) return null
+
+        fun calculateNumStep(percentageStep: Double): Int {
+            val numSteps = round(100 / percentageStep).toInt()
+            if (numSteps <= 10) return numSteps
+            if (numSteps % 10 == 0) return 10
+            return calculateNumStep(percentageStep * 2)
+        }
+
+        return calculateNumStep(((attributes as Map<*, *>)["percentage_step"] as? Double)?.toDouble() ?: 1.0) - 1
+    } catch (e: Exception) {
+        Log.e(EntityExt.TAG, "Unable to get getFanPercentageStep")
```

Comment:
This should match the function name

```suggestion
        Log.e(EntityExt.TAG, "Unable to get getFanSteps")
```

Reply:
Oops, I changed the function name at some point and didn't think about it anymore.